integrationTests/vm/esdt/esdtNft_test.go

```diff
@@ -0,0 +1,324 @@
+	nrRoundsToPropagateMultiShard = 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	checkNftData(t, nodes[1].OwnAccount.Address, nodes, tokenIdentifier, &nftMetaData, false)
+
+	// burn quantity
+	quantityToBurn := int64(4)
+	quantityToBurnArg := hex.EncodeToString(big.NewInt(quantityToBurn).Bytes())
+	txData = []byte(core.BuiltInFunctionESDTNFTBurn + "@" + hex.EncodeToString([]byte(tokenIdentifier)) +
+		"@" + nonceArg + "@" + quantityToBurnArg)
+	integrationTests.CreateAndSendTransaction(
+		nodes[1],
+		nodes,
+		big.NewInt(0),
+		nodes[1].OwnAccount.Address,
+		string(txData),
+		integrationTests.AdditionalGasLimit,
+	)
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard = 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	nftMetaData.quantity -= quantityToBurn
+	checkNftData(t, nodes[1].OwnAccount.Address, nodes, tokenIdentifier, &nftMetaData, false)
+}
+
+func issueNFT(nodes []*integrationTests.TestProcessorNode, esdtType string) {
+	ticker := "SFT"
+	tokenName := "token"
+	issuePrice := big.NewInt(1000)
+
+	tokenIssuer := nodes[0]
+
+	txData := txDataBuilder.NewBuilder()
+
+	issueFunc := "issueNonFungible"
+	if esdtType == core.SemiFungibleESDT {
+		issueFunc = "issueSemiFungible"
+	}
+	txData.Clear().Func(issueFunc).Str(tokenName).Str(ticker)
+	txData.CanFreeze(false).CanWipe(false).CanPause(false)
+
+	integrationTests.CreateAndSendTransaction(tokenIssuer, nodes, issuePrice, vm.ESDTSCAddress, txData.ToString(), core.MinMetaTxExtraGasCost)
+}
+
+type nftArguments struct {
+	name       []byte
+	quantity   int64
+	royalties  int64
+	hash       []byte
+	attributes []byte
+	uri        [][]byte
+}
+
+func createNFT(tokenIdentifier []byte, issuer *integrationTests.TestProcessorNode, nodes []*integrationTests.TestProcessorNode, args *nftArguments) {
+	txData := fmt.Sprintf("%s@%s@%s@%s@%s@%s@%s@%s@",
+		core.BuiltInFunctionESDTNFTCreate,
+		hex.EncodeToString(tokenIdentifier),
+		hex.EncodeToString(big.NewInt(args.quantity).Bytes()),
+		hex.EncodeToString(args.name),
+		hex.EncodeToString(big.NewInt(args.royalties).Bytes()),
+		hex.EncodeToString(args.hash),
+		hex.EncodeToString(args.attributes),
+		hex.EncodeToString(args.uri[0]),
+	)
+
+	integrationTests.CreateAndSendTransaction(issuer, nodes, big.NewInt(0), issuer.OwnAccount.Address, txData, integrationTests.AdditionalGasLimit)
+}
+
+func checkNftData(
+	t *testing.T,
+	address []byte,
+	nodes []*integrationTests.TestProcessorNode,
+	tokenName string,
+	args *nftArguments,
+	shouldNotFound bool,
+) {
+	tokenIdentifierPlusNonce := []byte(tokenName)
+	tokenIdentifierPlusNonce = append(tokenIdentifierPlusNonce, big.NewInt(0).SetUint64(1).Bytes()...)
+	esdtData := getESDTTokenData(t, address, nodes, string(tokenIdentifierPlusNonce))
+
+	if shouldNotFound {
```

Comment:
I think this bool is not needed. you can use args.quantity == 0

Reply:
done. nice idea :+1: